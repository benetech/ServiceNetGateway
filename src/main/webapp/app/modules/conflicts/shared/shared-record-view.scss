@import '../../../app-variables';
@import '../../../bootstrap-variables';

.title {
  margin-top: 15px;
}

.blueBorder {
  border-color: $blue-card-text;
  border-width: 3px;
}

.grayBorder {
  border-color: $gray-card-accent;
  border-width: 3px;
}

.icon-conflicting {
  color: $blue-card-text;
}

.icon-excluded {
  color: $gray-card-accent;
}

.icon-text {
  margin-top: 8px;
  margin-left: -45px;
  position: relative;
}

.icon-textarea {
  margin-top: 8px;
  margin-left: -45px !important;
  position: relative;
}

.icon-checkbox {
  margin-left: 5px;
}

.tooltip-inner {
  color: $tooltip-text-secondary;
  background-color: $tooltip-bg;
  box-shadow: 2px 2px 2px $tooltip-text-secondary;
}

.tooltip {
  margin-left: -10px;
}

.tooltip.show {
  opacity: 1 !important;
}

.half-rule {
  margin-top: 8px;
  margin-bottom: 8px;
  background-color: $tooltip-text-secondary;
}

.suggested {
  color: $tooltip-text-primary;
  text-align: left;
}

.secondary {
  color: $tooltip-text-secondary;
  margin: 0;
}

.jumbotron {
  margin-top: 15px;
  padding: 0;
  background-color: transparent;
  border: 0;
}

.jumbotron-header {
  padding: 10px;
  margin: 0;
  background-color: $compare-header-bg;
  border: 1px $compare-border solid;
  border-bottom: 0;
}

.jumbotron-content {
  padding: 10px;
  margin: 0;
  background-color: $compare-content-bg;
  border: 1px $compare-border solid;
  max-height: 460px;
  overflow-y: scroll;
}

.jumbotron-header-title {
  color: rgb(109, 108, 108); //secondary
}

.jumbotron-header-text {
  margin: 0;
}

.review-title {
  margin-bottom: 0;
}

.collapseBtn {
  display: inline;
  cursor: pointer;
}

.expandBtn {
  margin-top: 15px;
  cursor: pointer;
}

.collapseIcon {
  display: inline;
  margin-right: 5px;
}

.chainIcon {
  display: inline;
  margin-left: 5px;
}

.matchToggle {
  float: right;
}

.changeRecordSelect {
  padding-left: 0px;
}

.copy-to-clipboard-icon {
  position: absolute;
  right: 32px;
  margin-top: 6px;
  cursor: pointer;
  visibility: hidden;
  opacity: 0;
  transition: visibility 0s, opacity 0.5s linear;
  color: $gray-card-accent;
  padding: 0 5px;
}

.url-link {
  position: absolute;
  left: 25px;
  margin-top: 6px;
  cursor: pointer;
  visibility: visible;
  color: $blue-card-text;
}

.url-icon {
  margin-right: 10px;
}

.input-field-container {
  &:hover {
    .copy-to-clipboard-icon {
      visibility: visible;
      opacity: 1;
    }
  }
}

.tooltip-clip-inner {
  color: white !important;
  background-color: $tooltip-dark-bg !important;
}

.tooltip-clip {
  top: 5px !important;
  .arrow::before {
    border-bottom-color: $tooltip-dark-bg !important;
  }
}

span[role='button'] {
  padding: 5px;
  cursor: pointer;
}

.text-blue {
  color: #158cba;
}

.back-to-top {
  div:first-child {
    margin-right: 10px;
    text-align: center;
    height: 100px;
    width: 100px;
    border-radius: 100px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    &:hover {
      background-color: $transparent005;
    }
  }
}

.dismiss-modal {
  background-color: $modal-dark;
  color: $white;

  .close {
    color: $white;
  }

  .modal-header {
    border: none;
  }

  .modal-footer {
    border: none;
  }
}

.close-button,
.close-button:hover,
.close-button:active,
.close-button:focus {
  background-color: transparent;
  color: $white;
  border: none;
  box-shadow: none;
}

.taxonomy-pills {
  font-size: 1.25em;
  span {
    margin-right: 0.5em;
  }
}
