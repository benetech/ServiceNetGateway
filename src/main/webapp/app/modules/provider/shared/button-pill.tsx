import React from 'react';
import { Translate } from 'react-jhipster';

const ButtonPill = props => (
  <div className="button-pill" onClick={props.onClick}>
    <div className={`d-inline ${props.additionalClass}`}>
      <b>
        <Translate contentKey={props.translate} />
      </b>
    </div>
  </div>
);

export default ButtonPill;
