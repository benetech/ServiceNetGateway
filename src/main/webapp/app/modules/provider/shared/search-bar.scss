@import '../app-variables';
@import '../filter-card';

.search-from {
  background-color: white;
}

.search-from {
  height: 30px;
  display: flex;
  padding: 2px 4px;
  box-shadow: 1px 1px 2px 1px rgba(0, 0, 0, 0.3);
  align-items: center;
  border-radius: 25px;
}

.search-input-container {
  flex: 1;
  margin-left: 8px;
}

input[type='search']::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.search-input {
  outline: none;
  font: inherit;
  color: currentColor;
  width: 100%;
  border: 0;
  height: 1.1876em;
  margin: 0;
  display: block;
  padding: 6px 0 7px;
  min-width: 0;
  background: none;
  box-sizing: content-box;
  animation-name: mui-auto-fill-cancel;
  -webkit-tap-highlight-color: transparent;
}

.search-button {
  flex: 0 0 auto;
  color: rgba(0, 0, 0, 0.54);
  padding: 8px;
  overflow: visible;
  font-size: 1.5rem;
  text-align: center;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  background: rgba(255, 255, 255, 0);
  border: none;
}

.search-icon {
  width: 100%;
  display: flex;
  align-items: inherit;
  justify-content: inherit;
}

@media only screen and (max-width: 768px) {
  .search-bar {
    padding-right: 0;
    margin-bottom: 0.5rem;
    margin-top: 0.25rem;

    .search-from {
      width: 100%;
    }
  }
}
