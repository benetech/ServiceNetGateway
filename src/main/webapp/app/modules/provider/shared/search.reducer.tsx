export const ACTION_TYPES = {
  SET_TEXT: 'search/SET_TEXT',
  RESET_TEXT: 'search/RESET_TEXT'
};

export const initialState = {
  text: ''
};

export type SearchState = Readonly<typeof initialState>;

export default (state: SearchState = initialState, action): SearchState => {
  switch (action.type) {
    case ACTION_TYPES.SET_TEXT:
      return {
        ...state,
        text: action.text
      };
    case ACTION_TYPES.RESET_TEXT:
      return {
        ...state,
        text: initialState.text
      };
    default:
      return state;
  }
};

export const setText = text => dispatch => {
  dispatch({
    type: ACTION_TYPES.SET_TEXT,
    text
  });
};

export const resetText = () => dispatch => {
  dispatch({
    type: ACTION_TYPES.RESET_TEXT
  });
};
