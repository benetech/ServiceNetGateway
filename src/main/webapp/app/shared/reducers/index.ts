import { combineReducers } from 'redux';
import { loadingBarReducer as loadingBar } from 'react-redux-loading-bar';

import locale, { LocaleState } from './locale';
import authentication, { AuthenticationState } from './authentication';
import applicationProfile, { ApplicationProfileState } from './application-profile';

import administration, { AdministrationState } from 'app/modules/administration/administration.reducer';
import scheduler, { SchedulerState } from 'app/modules/administration/scheduler/scheduler.reducer';
import userManagement, { UserManagementState } from 'app/modules/administration/user-management/user-management.reducer';
import register, { RegisterState } from 'app/modules/account/register/register.reducer';
import activate, { ActivateState } from 'app/modules/account/activate/activate.reducer';
import password, { PasswordState } from 'app/modules/account/password/password.reducer';
import settings, { SettingsState } from 'app/modules/account/settings/settings.reducer';
import passwordReset, { PasswordResetState } from 'app/modules/account/password-reset/password-reset.reducer';
// prettier-ignore
import systemAccount, {
SystemAccountState
 } from 'app/entities/system-account/system-account.reducer';
// prettier-ignore
import documentUpload, {
DocumentUploadState
 } from 'app/entities/document-upload/document-upload.reducer';
// prettier-ignore
import organization, {
OrganizationState
 } from 'app/entities/organization/organization.reducer';
// prettier-ignore
import service, {
ServiceState
 } from 'app/entities/service/service.reducer';
// prettier-ignore
import program, {
ProgramState
 } from 'app/entities/program/program.reducer';
// prettier-ignore
import serviceAtLocation, {
ServiceAtLocationState
 } from 'app/entities/service-at-location/service-at-location.reducer';
// prettier-ignore
import location, {
LocationState
 } from 'app/entities/location/location.reducer';
// prettier-ignore
import physicalAddress, {
PhysicalAddressState
 } from 'app/entities/physical-address/physical-address.reducer';
// prettier-ignore
import postalAddress, {
PostalAddressState
 } from 'app/entities/postal-address/postal-address.reducer';
// prettier-ignore
import phone, {
PhoneState
 } from 'app/entities/phone/phone.reducer';
// prettier-ignore
import contact, {
ContactState
 } from 'app/entities/contact/contact.reducer';
// prettier-ignore
import regularSchedule, {
RegularScheduleState
 } from 'app/entities/regular-schedule/regular-schedule.reducer';
// prettier-ignore
import holidaySchedule, {
HolidayScheduleState
 } from 'app/entities/holiday-schedule/holiday-schedule.reducer';
// prettier-ignore
import funding, {
FundingState
 } from 'app/entities/funding/funding.reducer';
// prettier-ignore
import eligibility, {
EligibilityState
 } from 'app/entities/eligibility/eligibility.reducer';
// prettier-ignore
import serviceArea, {
ServiceAreaState
 } from 'app/entities/service-area/service-area.reducer';
// prettier-ignore
import requiredDocument, {
RequiredDocumentState
 } from 'app/entities/required-document/required-document.reducer';
// prettier-ignore
import paymentAccepted, {
PaymentAcceptedState
} from 'app/entities/payment-accepted/payment-accepted.reducer';
// prettier-ignore
import language, {
LanguageState
 } from 'app/entities/language/language.reducer';
// prettier-ignore
import accessibilityForDisabilities, {
AccessibilityForDisabilitiesState
 } from 'app/entities/accessibility-for-disabilities/accessibility-for-disabilities.reducer';
// prettier-ignore
import serviceTaxonomy, {
ServiceTaxonomyState
 } from 'app/entities/service-taxonomy/service-taxonomy.reducer';
// prettier-ignore
import taxonomy, {
TaxonomyState
 } from 'app/entities/taxonomy/taxonomy.reducer';
// prettier-ignore
import organizationMatch, {
OrganizationMatchState
 } from 'app/entities/organization-match/organization-match.reducer';
// prettier-ignore
import metadata, {
MetadataState
} from 'app/entities/metadata/metadata.reducer';
// prettier-ignore
import openingHours, {
  OpeningHoursState
} from 'app/entities/opening-hours/opening-hours.reducer';
// prettier-ignore
import uploadPage, {
  UploadPageState
} from 'app/modules/upload/upload-page.reducer';
// prettier-ignore
import sharedRecordView, {
  SharedRecordViewState
} from 'app/modules/conflicts/shared/shared-record-view.reducer';
// prettier-ignore
import conflict, {
  ConflictState
} from 'app/entities/conflict/conflict.reducer';
// prettier-ignore
import dataImportReport, {
  DataImportReportState
} from 'app/entities/data-import-report/data-import-report.reducer';
// prettier-ignore
import activity, {
  ActivityState
} from 'app/shared/reducers/activity.reducer';
// prettier-ignore
import fieldExclusion, {
  FieldExclusionState
} from 'app/entities/field-exclusion/field-exclusion.reducer';
// prettier-ignore
import exclusionsConfig, {
  ExclusionsConfigState
} from 'app/entities/exclusions-config/exclusions-config.reducer';
// prettier-ignore
import geocodingResult, {
  GeocodingResultState
} from 'app/entities/geocoding-result/geocoding-result.reducer';
import filterActivity, { FilterActivityState } from 'app/modules/home/filter-activity.reducer';
// prettier-ignore
import filterShelter, { FilterShelterState } from 'app/modules/shelter/filter-shelter.reducer';
// prettier-ignore
import beds, {
  BedsState
} from 'app/entities/beds/beds.reducer';
// prettier-ignore
import shelter, {
  ShelterState
} from 'app/entities/shelter/shelter.reducer';
// prettier-ignore
import option, {
  OptionState
} from 'app/entities/option/option.reducer';
// prettier-ignore
import locationExclusion, {
  LocationExclusionState
} from 'app/entities/location-exclusion/location-exclusion.reducer';
// prettier-ignore
import organizationError, {
  OrganizationErrorState
} from 'app/entities/organization-error/organization-error.reducer';
import matchSimilarity, { MatchSimilarityState } from 'app/entities/match-similarity/match-similarity.reducer';
// prettier-ignore
import activityFilter, {
  ActivityFilterState
} from 'app/entities/activity-filter/activity-filter.reducer';
// prettier-ignore
import taxonomyGroup, {
  TaxonomyGroupState
} from 'app/entities/taxonomy-group/taxonomy-group.reducer';
// prettier-ignore
import fieldsDisplaySettings, {
  FieldsDisplaySettingsState
} from 'app/entities/fields-display-settings/fields-display-settings.reducer';
// prettier-ignore
import locationFieldsValue, {
  LocationFieldsValueState
} from 'app/entities/location-fields-value/location-fields-value.reducer';
// prettier-ignore
import organizationFieldsValue, {
  OrganizationFieldsValueState
} from 'app/entities/organization-fields-value/organization-fields-value.reducer';
// prettier-ignore
import physicalAddressFieldsValue, {
  PhysicalAddressFieldsValueState
} from 'app/entities/physical-address-fields-value/physical-address-fields-value.reducer';
// prettier-ignore
import postalAddressFieldsValue, {
  PostalAddressFieldsValueState
} from 'app/entities/postal-address-fields-value/postal-address-fields-value.reducer';
// prettier-ignore
import serviceFieldsValue, {
  ServiceFieldsValueState
} from 'app/entities/service-fields-value/service-fields-value.reducer';
// prettier-ignore
import serviceTaxonomiesDetailsFieldsValue, {
  ServiceTaxonomiesDetailsFieldsValueState
} from 'app/entities/service-taxonomies-details-fields-value/service-taxonomies-details-fields-value.reducer';
// prettier-ignore
import contactDetailsFieldsValue, {
  ContactDetailsFieldsValueState
} from 'app/entities/contact-details-fields-value/contact-details-fields-value.reducer';
// prettier-ignore
import feedback, { FeedbackState } from 'app/modules/feedback/feedback.reducer';
import requestLogger, { RequestLoggerState } from 'app/entities/request-logger/request-logger.reducer';
import clientManagement, { ClientManagementState } from 'app/modules/administration/client-management/client-management.reducer';
import dataStatus, { DataStatusState } from 'app/modules/data-status/data-status.reducer';
import providerRecord, { ProviderRecordsState } from 'app/modules/provider/provider-record.reducer';
import providerFilter, { ProviderFilterState } from 'app/modules/provider/provider-filter.reducer';
import deactivatedRecords, { DeactivatedRecordsState } from 'app/modules/provider/deactivated/deactivated-records.reducer';
import usersOrganizations, { UsersOrganizationsState } from 'app/modules/administration/users-organizations/users-organizations.reducer';

// prettier-ignore
import dailyUpdate, {
  DailyUpdateState
} from 'app/entities/daily-update/daily-update.reducer';

// prettier-ignore
import search, { SearchState } from 'app/modules/provider/menus/search.reducer';

// prettier-ignore
import silo, {
  SiloState
} from 'app/entities/silo/silo.reducer';
// prettier-ignore
import userGroup, {
  UserGroupState
} from 'app/entities/user-group/user-group.reducer';
// prettier-ignore
import beneficiary, {
  BeneficiaryState
} from 'app/entities/beneficiary/beneficiary.reducer';
// prettier-ignore
import referral, {
  ReferralState
} from 'app/entities/referral/referral.reducer';
/* jhipster-needle-add-reducer-import - JHipster will add reducer here */

export interface IRootState {
  readonly authentication: AuthenticationState;
  readonly clientManagement: ClientManagementState;
  readonly locale: LocaleState;
  readonly applicationProfile: ApplicationProfileState;
  readonly administration: AdministrationState;
  readonly scheduler: SchedulerState;
  readonly userManagement: UserManagementState;
  readonly register: RegisterState;
  readonly activate: ActivateState;
  readonly passwordReset: PasswordResetState;
  readonly password: PasswordState;
  readonly settings: SettingsState;
  readonly systemAccount: SystemAccountState;
  readonly documentUpload: DocumentUploadState;
  readonly organization: OrganizationState;
  readonly service: ServiceState;
  readonly program: ProgramState;
  readonly serviceAtLocation: ServiceAtLocationState;
  readonly location: LocationState;
  readonly physicalAddress: PhysicalAddressState;
  readonly postalAddress: PostalAddressState;
  readonly phone: PhoneState;
  readonly contact: ContactState;
  readonly regularSchedule: RegularScheduleState;
  readonly holidaySchedule: HolidayScheduleState;
  readonly funding: FundingState;
  readonly eligibility: EligibilityState;
  readonly serviceArea: ServiceAreaState;
  readonly requiredDocument: RequiredDocumentState;
  readonly paymentAccepted: PaymentAcceptedState;
  readonly language: LanguageState;
  readonly accessibilityForDisabilities: AccessibilityForDisabilitiesState;
  readonly serviceTaxonomy: ServiceTaxonomyState;
  readonly taxonomy: TaxonomyState;
  readonly organizationMatch: OrganizationMatchState;
  readonly metadata: MetadataState;
  readonly openingHours: OpeningHoursState;
  readonly uploadPage: UploadPageState;
  readonly sharedRecordView: SharedRecordViewState;
  readonly conflict: ConflictState;
  readonly dataImportReport: DataImportReportState;
  readonly activity: ActivityState;
  readonly filterActivity: FilterActivityState;
  readonly fieldExclusion: FieldExclusionState;
  readonly exclusionsConfig: ExclusionsConfigState;
  readonly geocodingResult: GeocodingResultState;
  readonly beds: BedsState;
  readonly shelter: ShelterState;
  readonly option: OptionState;
  readonly locationExclusion: LocationExclusionState;
  readonly organizationError: OrganizationErrorState;
  readonly matchSimilarity: MatchSimilarityState;
  readonly activityFilter: ActivityFilterState;
  readonly taxonomyGroup: TaxonomyGroupState;
  readonly fieldsDisplaySettings: FieldsDisplaySettingsState;
  readonly locationFieldsValue: LocationFieldsValueState;
  readonly organizationFieldsValue: OrganizationFieldsValueState;
  readonly physicalAddressFieldsValue: PhysicalAddressFieldsValueState;
  readonly postalAddressFieldsValue: PostalAddressFieldsValueState;
  readonly serviceFieldsValue: ServiceFieldsValueState;
  readonly serviceTaxonomiesDetailsFieldsValue: ServiceTaxonomiesDetailsFieldsValueState;
  readonly contactDetailsFieldsValue: ContactDetailsFieldsValueState;
  readonly dailyUpdate: DailyUpdateState;
  readonly silo: SiloState;
  readonly userGroup: UserGroupState;
  readonly beneficiary: BeneficiaryState;
  readonly referral: ReferralState;
  /* jhipster-needle-add-reducer-type - JHipster will add reducer type here */
  readonly loadingBar: any;
  readonly filterShelter: FilterShelterState;
  readonly feedback: FeedbackState;
  readonly dataStatus: DataStatusState;
  readonly requestLogger: RequestLoggerState;
  readonly providerRecord: ProviderRecordsState;
  readonly providerFilter: ProviderFilterState;
  readonly search: SearchState;
  readonly deactivatedRecords: DeactivatedRecordsState;
  readonly usersOrganizations: UsersOrganizationsState;
}

const rootReducer = combineReducers<IRootState>({
  authentication,
  clientManagement,
  locale,
  applicationProfile,
  administration,
  scheduler,
  userManagement,
  register,
  activate,
  passwordReset,
  password,
  settings,
  systemAccount,
  documentUpload,
  organization,
  service,
  program,
  serviceAtLocation,
  location,
  physicalAddress,
  postalAddress,
  phone,
  contact,
  regularSchedule,
  holidaySchedule,
  funding,
  eligibility,
  serviceArea,
  requiredDocument,
  paymentAccepted,
  language,
  accessibilityForDisabilities,
  serviceTaxonomy,
  taxonomy,
  organizationMatch,
  metadata,
  openingHours,
  uploadPage,
  sharedRecordView,
  conflict,
  dataImportReport,
  activity,
  fieldExclusion,
  exclusionsConfig,
  geocodingResult,
  beds,
  shelter,
  option,
  locationExclusion,
  organizationError,
  matchSimilarity,
  activityFilter,
  taxonomyGroup,
  fieldsDisplaySettings,
  locationFieldsValue,
  organizationFieldsValue,
  physicalAddressFieldsValue,
  postalAddressFieldsValue,
  serviceFieldsValue,
  serviceTaxonomiesDetailsFieldsValue,
  contactDetailsFieldsValue,
  dailyUpdate,
  silo,
  userGroup,
  beneficiary,
  referral,
  /* jhipster-needle-add-reducer-combine - JHipster will add reducer here */
  loadingBar,
  filterActivity,
  filterShelter,
  feedback,
  requestLogger,
  dataStatus,
  providerRecord,
  providerFilter,
  search,
  deactivatedRecords,
  usersOrganizations
});

export default rootReducer;
