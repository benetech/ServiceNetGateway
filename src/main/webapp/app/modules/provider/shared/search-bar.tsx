import React from 'react';
import '../provider-shared.scss';
import './search-bar.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { translate } from 'react-jhipster';
import { connect } from 'react-redux';

export interface ISearchBarProp extends StateProps, DispatchProps {
  onSwitchFocus?: any;
  onSearch: any;
  onReset: any;
  width?: string;
}

export interface ISearchBarState {
  text: string;
}

export class SearchBar extends React.Component<ISearchBarProp, ISearchBarState> {
  constructor(props) {
    super(props);
    this.state = {
      text: ''
    };
  }

  componentDidMount() {
    this.props.onReset();
  }

  updateText = event => {
    this.setState({
      text: event.target.value
    });
  };

  search = event => {
    if (event !== null) {
      event.preventDefault();
    }
    this.props.onSearch(this.state.text);
  };

  reset = () => {
    this.setState({
      text: ''
    });
    this.props.onReset();
  };

  onFocus = e => {
    if (!e.currentTarget.contains(e.relatedTarget)) {
      if (this.props.onSwitchFocus) {
        this.props.onSwitchFocus(true);
      }
    }
  };

  onBlur = e => {
    if (!e.currentTarget.contains(e.relatedTarget)) {
      if (this.props.onSwitchFocus) {
        this.props.onSwitchFocus(false);
      }
    }
  };

  render() {
    const { width } = this.props;
    return (
      <div className="search-bar">
        <form className="search-from" style={{ width: width ? width : null }} onSubmit={this.search}>
          <div className="search-button">
            <span className="search-button search-icon" onClick={this.search}>
              <FontAwesomeIcon className="self-align-center" size="xs" icon="search" />
            </span>
          </div>
          <div className="search-input-container">
            <input
              id="search"
              className="search-input"
              type="search"
              value={this.state.text}
              placeholder={translate('providerSite.searchPlaceholder')}
              onChange={this.updateText}
              onFocus={this.onFocus}
              onBlur={this.onBlur}
            />
            <label className="sr-only" htmlFor="search">
              {translate('providerSite.searchPlaceholder')}
            </label>
          </div>
          <div className="search-button">
            <span className="search-button search-icon" onClick={this.reset}>
              <FontAwesomeIcon className="self-align-center" size="xs" icon="times" />
            </span>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({});

const mapDispatchToProps = {};

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  null,
  mapDispatchToProps
)(SearchBar);
