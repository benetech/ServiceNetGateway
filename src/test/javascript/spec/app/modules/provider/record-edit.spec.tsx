import React from 'react';
import RecordEdit from 'app/modules/provider/record/record-edit';
import { store } from './shared/data';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

describe('Record edit', () => {
  let mountedWrapper;

  it('renders correctly', () => {
    const props = {
      match: {
        params: {
          id: 123
        }
      }
    };
    mountedWrapper = mount(
      <Provider store={store}>
        <Router>
          // @ts-ignore
          <RecordEdit {...props} />
        </Router>
      </Provider>
    );
    expect(mountedWrapper).toMatchSnapshot();
  });
});
