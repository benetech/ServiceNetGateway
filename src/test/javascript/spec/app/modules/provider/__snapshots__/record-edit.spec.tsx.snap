// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record edit renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      // @ts-ignore
      <Connect(RecordEdit)
        match={
          Object {
            "params": Object {
              "id": 123,
            },
          }
        }
      >
        <RecordEdit
          deactivateEntity={[Function]}
          getProviderEntity={[Function]}
          getProviderTaxonomies={[Function]}
          locations={Array []}
          match={
            Object {
              "params": Object {
                "id": 123,
              },
            }
          }
          organization={
            Object {
              "dailyUpdates": Array [],
              "description": "fakeDesc",
              "email": "fakeEmail",
              "id": "123",
              "locations": Array [],
              "name": "fakeName",
              "services": Array [],
              "updatedAt": "fakeDate",
              "url": "fakeUrl",
            }
          }
          taxonomyOptions={Array []}
          updateSuccess={true}
          updateUserOwnedEntity={[Function]}
          updating={true}
        />
      </Connect(RecordEdit)>
    </Router>
  </HashRouter>
</Provider>
`;
